modules/sdk-core/src/bitgo/wallet/wallet.ts

```diff
@@ -1583,6 +1584,38 @@ export class Wallet implements IWallet {
     return prebuild as PrebuildTransactionResult;
   }
 
+  /**
+   * Signs a UTXO coin transaction/psbt using external signer.
+   */
+  private async signTransactionUtxoExternalSigner(
+    customSigningFunction: CustomSigningFunction,
+    signTransactionParams: { txPrebuild: TransactionPrebuild; pubs?: string[]; coin: IBaseCoin }
+  ) {
+    const getTxHex = (v: SignedTransaction): string => {
+      if ('txHex' in v) {
+        return v.txHex;
+      }
+      throw new Error('txHex not found in signTransaction result');
+    };
+
+    const signerNonceTx = await customSigningFunction({
+      ...signTransactionParams,
+      signingStep: 'signerNonce',
+    });
+
+    const cosignerNonceTx = await this.baseCoin.signTransaction({
+      ...signTransactionParams,
+      txPrebuild: { ...signTransactionParams.txPrebuild, txHex: getTxHex(signerNonceTx) },
+      signingStep: 'cosignerNonce',
+    });
+
+    return await customSigningFunction({
+      ...signTransactionParams,
+      txPrebuild: { ...signTransactionParams.txPrebuild, txHex: getTxHex(cosignerNonceTx) },
+      signingStep: 'signerSignature',
+    });
+  }
```

Comment:
let's move this to AbstractUtxoCoin `signWithCustomSigningFunction`

Reply:
Done